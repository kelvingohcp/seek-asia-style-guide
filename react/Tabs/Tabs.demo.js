import Tabs from './Tabs';
import * as sketch from './Tabs.sketch';
import React from 'react';
import PropTypes from 'prop-types';
import JobCard from '../JobCard/JobCard';
import PageBlock from '../PageBlock/PageBlock';

const TabsContainer = ({ component: DemoComponent, componentProps }) => {
  return (
    <PageBlock style={{ width: '100%' }}>
      <DemoComponent {...componentProps} />
    </PageBlock>
  );
};

TabsContainer.propTypes = {
  component: PropTypes.func,
  componentProps: PropTypes.object
};

export default {
  route: '/tabs',
  title: 'Tabs',
  component: Tabs,
  sketch,
  container: TabsContainer,
  initialProps: {
    items: [
      {
        title: 'Job Details',
        content: (
          <JobCard
            job={{
              company: {
                name: 'SEEK Asia'
              },
              isExpired: false,
              jobUrl: 'https://www-dev.jobstreet.com.my/en/job/20171002-3-senior-front-end-developer-update-x-2-6100835/origin/dev/sources/3?fr=J',
              jobTitle: 'Senior Software Engineer (6 months Contract)',
              locations: [
                {
                  link: '/jobCard',
                  name: 'Pahang'
                },
                {
                  child: {
                    child: {
                      name: 'Near Leisure Mall'
                    },
                    link: '/jobCard',
                    name: 'Cheras'
                  },
                  link: '/jobCard',
                  name: 'Selangor'
                }
              ],
              postingDuration: '1 hour ago',
              salary: 'RM99999 - RM999999',
              sellingPoints: [
                'We practice a vibrant & energetic office culture',
                'Our company supports a fun yet balanced working environment',
                'We support a safe environment for our employees'
              ]
            }}
            jobAdType="jobsdbDefault"
          />
        )
      },
      {
        title: 'Company Info',
        content: <div>Company info</div>
      }
    ]
  },
  options: []
};
