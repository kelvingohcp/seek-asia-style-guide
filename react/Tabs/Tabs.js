import styles from './Tabs.less';
import React from 'react';
import PropTypes from 'prop-types';
import {
  Tab as ReactTab,
  Tabs as ReactTabs,
  TabList as ReactTabList,
  TabPanel as ReactTabPanel
} from 'react-tabs';
import Text from '../Text/Text';
import classnames from 'classnames';

export default function Tabs({ items, className }) {
  const tabsTitles = [];
  const tabsContent = [];
  items.forEach(function(item, index) {
    if (item.title) {
      tabsTitles.push(
        <ReactTab
          className={styles.tab}
          selectedClassName={styles.selectedTab}
          key={index}>
          <Text waving strong>
            {item.title}
          </Text>
        </ReactTab>
      );
      tabsContent.push(
        <ReactTabPanel key={index}>
          {item.content}
        </ReactTabPanel>
      );
    }
  });
  return (
    <ReactTabs className={classnames(styles.root, className)}>
      <ReactTabList className={styles.tabList}>{tabsTitles}</ReactTabList>
      {tabsContent}
    </ReactTabs>
  );
}

Tabs.propTypes = {
  items: PropTypes.array.isRequired,
  className: PropTypes.string
};
