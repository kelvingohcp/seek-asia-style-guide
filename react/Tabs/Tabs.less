@import (reference) '~seek-asia-style-guide/theme';

.root {
  width: 100%;
}

.tabList {
  display: inline-flex;
  justify-content: flex-start;
}

.tab {
  cursor: pointer;
  padding: @grid-row-height @grid-gutter-width;
}

.selectedTab {
  color: @actionBlue;
  border-bottom-style: solid;
  border-bottom-color: @actionBlue solid;
  border-bottom-width: @grid-row-height * 2;
}
