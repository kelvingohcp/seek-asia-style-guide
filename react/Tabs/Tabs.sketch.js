import React from 'react';
import Tabs from './Tabs';

export const symbols = {
  '2 Tabs': (
    <Tabs
      items={[{ title: 'Tab 1', content: '' }, { title: 'Tab 2', content: '' }]}
    />
  )
};
