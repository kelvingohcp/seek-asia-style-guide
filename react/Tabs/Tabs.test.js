import { shallow } from 'enzyme';
import React from 'react';
import Tabs from './Tabs';

describe('Tabs', () => {
  it('should render with defaults', () => {
    expect(shallow(<Tabs items={[]} />)).toMatchSnapshot();
  });

  it('should render with className', () => {
    expect(shallow(<Tabs className="foo" />)).toMatchSnapshot();
  });

  it('should render with content', () => {
    expect(
      shallow(
        <Tabs
          content={[
            { title: 'Title 1', content: 'Content 1' },
            { title: 'Title 2', content: 'Content 2' }
          ]}
        />
      )
    ).toMatchSnapshot();
  });

  it('should render with no content', () => {
    expect(
      shallow(<Tabs content={[{ title: 'Title 1' }, { title: 'Title 2' }]} />)
    ).toMatchSnapshot();
  });
});
