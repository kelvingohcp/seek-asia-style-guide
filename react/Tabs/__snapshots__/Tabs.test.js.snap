// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render with className 1`] = `
<Tabs
  className="root foo"
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  selectedIndex={null}
>
  <TabList
    className="tabList"
  />
</Tabs>
`;

exports[`Tabs should render with content 1`] = `
<Tabs
  className="root"
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  selectedIndex={null}
>
  <TabList
    className="tabList"
  >
    <Tab
      className="tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      key="0"
      panelId={null}
      selected={false}
      selectedClassName="selectedTab"
    >
      <Text
        baseline={true}
        strong={true}
        waving={true}
      >
        Title 1
      </Text>
    </Tab>
    <Tab
      className="tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      key="1"
      panelId={null}
      selected={false}
      selectedClassName="selectedTab"
    >
      <Text
        baseline={true}
        strong={true}
        waving={true}
      >
        Title 2
      </Text>
    </Tab>
  </TabList>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    key="0"
    selectedClassName="react-tabs__tab-panel--selected"
  >
    Content 1
  </TabPanel>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    key="1"
    selectedClassName="react-tabs__tab-panel--selected"
  >
    Content 2
  </TabPanel>
</Tabs>
`;

exports[`Tabs should render with defaults 1`] = `
<Tabs
  className="root"
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  selectedIndex={null}
>
  <TabList
    className="tabList"
  />
</Tabs>
`;

exports[`Tabs should render with no content 1`] = `
<Tabs
  className="root"
  defaultFocus={false}
  defaultIndex={null}
  forceRenderTabPanel={false}
  selectedIndex={null}
>
  <TabList
    className="tabList"
  >
    <Tab
      className="tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      key="0"
      panelId={null}
      selected={false}
      selectedClassName="selectedTab"
    >
      <Text
        baseline={true}
        strong={true}
        waving={true}
      >
        Title 1
      </Text>
    </Tab>
    <Tab
      className="tab"
      disabledClassName="react-tabs__tab--disabled"
      focus={false}
      id={null}
      key="1"
      panelId={null}
      selected={false}
      selectedClassName="selectedTab"
    >
      <Text
        baseline={true}
        strong={true}
        waving={true}
      >
        Title 2
      </Text>
    </Tab>
  </TabList>
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    key="0"
    selectedClassName="react-tabs__tab-panel--selected"
  />
  <TabPanel
    className="react-tabs__tab-panel"
    forceRender={false}
    key="1"
    selectedClassName="react-tabs__tab-panel--selected"
  />
</Tabs>
`;
